### PR TITLE
ENH: improve MinariDataset

### PR DESCRIPTION
**Pull request by younik**
# Description

Add `__getitem__`and `__len__()` to `MinariDataset` and few refactors.

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/tutorials/q_learning_torch.py
Comment (balisujohn) [reply docs/tutorials/q_learning_torch.py:169]: For reference, I think the code is not fully deterministic since I got:
````
Offline Q-Learning accumulated rew:  160.0
Dataset mean accumulated rew:  22.12796
````

**Review by balisujohn — COMMENTED** [review]
Seems to run fine and produces a performant policy, I requested a few minor changes, and asked a few questions.


One more comment: It might be good to provide a requirements.txt for tutorials(though the version of Minari this will depend on isn't in a release yet, so I suppose it would have to be in a later PR)

File: minari/dataset/minari_dataset.py
Comment (balisujohn) [inline minari/dataset/minari_dataset.py:385]: All changes look good, but please add  tests for `__iter__ `,`__getitem__` and `__len__`

File: docs/tutorials/q_learning_torch.py
Comment (Howuhh) [inline docs/tutorials/q_learning_torch.py:134]: I think it would be good to add shape annotations here, so that users can understand loss completely

File: docs/tutorials/q_learning_torch.py
Comment (younik) [reply docs/tutorials/q_learning_torch.py:142]: I think you are right as well; and yes, if I change it, it performs worse

File: docs/tutorials/q_learning_torch.py
Comment (younik) [reply docs/tutorials/q_learning_torch.py:154]: The initial idea of this tutorial was to show how to integrate the dataset in a common PyTorch pipeline bus, afaik, there is no paper showing this very simple method (that doesn't work in general). I wanted this to be as simple as possible

Maybe I should update the tutorial to reimplement a paper; it will be more complicated, but also more informative

File: docs/tutorials/q_learning_torch.py
Comment (younik) [reply docs/tutorials/q_learning_torch.py:134]: Good point, I will update

File: minari/dataset/minari_dataset.py
Comment (younik) [reply minari/dataset/minari_dataset.py:385]: Much easier to add after we merge https://github.com/Farama-Foundation/Minari/pull/101

File: docs/tutorials/q_learning_torch.py
Comment (balisujohn) [reply docs/tutorials/q_learning_torch.py:154]: If this method is custom and there is no paper describing it, I think that's fine.

File: docs/tutorials/behavioral_cloning.py
Comment (rodrigodelazcano) [inline docs/tutorials/behavioral_cloning.py:14]: "library"

File: docs/index.md
Comment (rodrigodelazcano) [inline docs/index.md:28]: I think we can include this tutorial in a new tab called "Using Datasets". What do you think?

**Review by rodrigodelazcano — CHANGES_REQUESTED** [review]
I just added some suggestions

File: docs/tutorials/behavioral_cloning.py
Comment (balisujohn) [inline docs/tutorials/behavioral_cloning.py:25]: maybe change "occurences" to "required packages"

File: docs/tutorials/behavioral_cloning.py
Comment (balisujohn) [inline docs/tutorials/behavioral_cloning.py:184]: this should say what a good reward is for the environment, I got 500 as the accumulated reward when I ran this(Which I assume is a good reward).

**Review by balisujohn — COMMENTED** [review]
Really minor change requests, and a reminder that we should add a version locked requirements.txt after the next minari release. Otherwise, looks good, and ready to merge, assuming 500 is a good amount of total reward for the final cloned policy.

File: docs/tutorials/behavioral_cloning.py
Comment (younik) [reply docs/tutorials/behavioral_cloning.py:184]: 500 is the maximum obtainable, since after the episode is truncated. I add the explanation

File: docs/index.md
Comment (younik) [reply docs/index.md:28]: Sounds good, done

**Review by balisujohn — APPROVED** [review]
LGTM :^)

**Review by rodrigodelazcano — APPROVED** [review]

**Comment by rodrigodelazcano** [comment]
Looks good! Ready to merge

**Pull request merged**
